### ISSUE TITLE
[Core] (Sub)ModelPart Name()

### ISSUE THREAD

**Issue by rubenzorrilla** [KratosMultiphysics/Kratos#6056]
When the name of a model part is retrieved, the method `Name()` always returns the value in `mName`. In case the model part of interest is a submodelpart, it returns the submodelpart name without the parent model part name prepending it.

After #6024 this might turn some warnings to errors. So far I've found one case in the FluidDynamicsApplication, but there could be more in other applications.

Having said this, I see two options:
- We keep the current behaviour of `Name()` and we implement a `FullName()` method.
- We modify the current behaviour of `Name()` to recursively check if the model part of interest is a submodelpart to prepend `mName` with the father model part model part name.

Personally, I'd rather go for the second one. However I haven't checked if there is any technical limitation.

**Comment by loumalouomega** [comment]
I prefer the first option, is back compatible. Many implementations already use Name and if we change it it will break a lo of code. Additionally I think is the proper way to proceed, as we may be interested only in the current model part name for some reason, and it is tedious to split names again.

**Comment by philbucher** [comment]
Very much agree with Vicente

**Comment by rubenzorrilla** [comment]
I though about the backward compatibility too... But as far as I understand there shouldn't be any problem as what can be retrieved right now with the partial name must be also possible to be retrieved with the full name.

My unique concerns are:
- When doing string comparisons of the model part names, something that I think is not a good practice as it limits extensibility.
- If there are two submodelparts with the same name but owned by different base model parts. Right now we have no control of which one is being retrieved (I'm saying this by heart so I'm not 100% sure).

**Comment by philbucher** [comment]
I think you are mixing things up... are you talking abt retrieving from Model or ModelPart?

> If there are two submodelparts with the same name but owned by different base model parts. Right now we have no control of which one is being retrieved (I'm saying this by heart so I'm not 100% sure).

This is the only reason for #6024

**Comment by rubenzorrilla** [comment]
Well, retrieving has to do with the name. Actually this is the case I found in the FluidDynamicsApplicatoin. In any case I figured out a way to circumvent it in that particular case but I feel it's still worth it to discuss it.

In any case, I'm not opposing to the `FullName` method, but I think it would have been nice to have a unique way to retrieve the name.

**Comment by philbucher** [comment]
+1 for adding FullName

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee agrees with `FullName`
